packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -595,6 +598,48 @@ export const RemoteServerFactory = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupSlasPrivateClientProxy(app, options) {
+        if (options.useSlasPrivateClient) {
+            const clientSecret = process?.env?.SLAS_PRIVATE_CLIENT_SECRET
+            if (!clientSecret) {
+                app.use('/ssr/auth', (_, res) => {
+                    return res.status(501).json({
+                        message:
+                            'Environment variable SLAS_PRIVATE_CLIENT_ID not set: LINK_TO_DOC HERE'
+                    })
+                })
+            } else {
+                const clientId = options.mobify.app.commerceAPI.parameters.clientId
+                const shortCode = options.mobify.app.commerceAPI.parameters.shortCode
+
+                const slasTarget = `https://${shortCode}.api.commercecloud.salesforce.com`
+
+                app.use(
+                    '/ssr/auth',
+                    createProxyMiddleware({
+                        target: slasTarget,
+                        changeOrigin: true,
+                        pathRewrite: {'/ssr/auth': ''},
+                        onProxyReq: (outGoingReq, incomingReq) => {
+                            if (incomingReq.path.match(/\/oauth2\/token/)) {
```

Comment:
We have this here because we do not want to apply the header replacement on calls to /oauth2/login since this endpoint uses the authorization header for a different purpose.

Reply:
That sounds like a good code comment!

Are there any other SLAS endpoints that use Authorization for something different?

In the default PWA Kit implementation, we call four SLAS endpoints:

```
/shopper/auth/v1/organizations/$ORG/oauth2/authorize
/shopper/auth/v1/organizations/$ORG/oauth2/login
/shopper/auth/v1/organizations/$ORG/oauth2/logout
/shopper/auth/v1/organizations/$ORG/oauth2/token
```

With private client, we shouldn't need to call `authorize` anymore ...

And of the remaining three, only `token` requires the client id / secret.

Could you think of any reason someone might need to customize the behaviour here?

Reply:
Of the 4 endpoints we use by default, only token requires the client id / secret, as you point out.

However, if projects were to customize and add, for example, passwordless login they would want to update the matcher to include `/oauth2/passwordless/token`

Reply:
I looked into SLAS' use of the `Authorization` header and aside from `/oauth2/login`, `/oauth2/trusted-agent/token` also uses this header for a different purpose.

All other SLAS endpoints aside from the two above either don't use the Authorization header or use it for the same reason as /oauth2/token.

Reply:
I did a more thorough look through SLAS to audit which endpoints use the client secret in their authorization header vs which endpoints use the authorization header for a different purpose:

Endpoints that use client secret in the Authorization header:

```
/oauth2/passwordless/login
/oauth2/token
/oauth2/session-bridge/token
/oauth2/trusted-system/token
/oauth2/password/action
/oauth2/passwordless/token
/oauth2/revoke
/oauth2/introspect
```

Endpoints that use Authorization header for something else:
```
/oauth2/login	(used for shopper credentials)
/oauth2/logout (Bearer - access token)
/oauth2/trusted-agent/token (Bearer - access token. client secret provided via the _sfdc_client_auth header instead)
/oauth2/userinfo (Bearer - access token)
```

Endpoints that don’t use a client secret / Authorization header:
```
/oauth2/authorize
/oauth2/trusted-agent/authorize
/oauth2/password/reset
/oauth2/.well-known/openid-configuration
/oauth2/jwks
```